mix.exs

```diff
@@ -20,7 +20,9 @@ defmodule TypedStruct.MixProject do
       # Docs
       name: "TypedStruct",
       docs: [
-        main: "TypedStruct",
+        extras: ["CHANGELOG.md", "README.md", "CONTRIBUTING.md"],
```

Comment:
Ok for the `CHANGELOG.md`, but I do not see the point for the `CONTRIBUTING.md` since it only concerns developers and not users of the library.

As for the `README.md`, it would indeed be good to avoid documentation duplication between the `@moduledoc` and the `README.md`, but your commit also removes all the details from the `@moduledoc`. What if someone wants to read it via `h` in `iex`? I think we could either import the `README.md` in the `@moduledoc` somehow, so the documentation still exist in the module, or find a way to generate the `README.md` from the `@moduledoc`.

Reply:
> Ok for the `CHANGELOG.md`, but I do not see the point for the `CONTRIBUTING.md` since it only concerns developers and not users of the library.

It's such a well-written contribution guideline, pity if this is not included the the final generated HTML doc. :smile: 

As mentioned in PR comment, the main purpose is to ensure the generated HTML doc will become the main source doc for this library. Instead of going through different files in GithHub, we can have quick overview (even search) of all relevant and combined docs for this library in HexDocs.pm.

> As for the `README.md`, it would indeed be good to avoid documentation duplication between the `@moduledoc` and the `README.md`, but your commit also removes all the details from the `@moduledoc`. What if someone wants to read it via `h` in `iex`? I think we could either import the `README.md` in the `@moduledoc` somehow, so the documentation still exist in the module, or find a way to generate the `README.md` from the `@moduledoc`.

I see. Definitely overlook this. I'll embed the readme portion into the module doc.